src/DatePicker.php

```diff
@@ -107,12 +107,16 @@ public function registerClientScript()
 
         $options = !empty($this->clientOptions) ? Json::encode($this->clientOptions) : '';
 
+        $js[] = "$selector.datepicker($options);";
+
         if ($this->inline) {
+            if (!empty($this->clientEvents['changeDate'])) {
+                $handler = $this->clientEvents['changeDate'];
+                $js[] = "$selector.on('changeDate', $handler);";
+            }
             $this->clientEvents['changeDate'] = "function (e){ jQuery('#$id').val(e.format());}";
         }
 
-        $js[] = "$selector.datepicker($options);";
-
         if (!empty($this->clientEvents)) {
```

Comment:
The following statement does exactly what you did above

Reply:
I moved this line above,
because $this->clientEvents array has duplicte keys "changeDate"